pkg/stream/consumer_batch.go

```diff
@@ -83,11 +84,20 @@ func (c *BatchConsumer) Run(kernelCtx context.Context) error {
 	return c.baseConsumer.run(kernelCtx, c.run)
 }
 
-func (c *BatchConsumer) run(ctx context.Context) error {
+func (c *BatchConsumer) run(ctx context.Context) (err error) {
 	logger := c.logger.WithContext(ctx)
+
 	defer logger.Debug("run is ending")
 	defer c.wg.Done()
-	defer c.processBatch(context.Background())
+	defer func() {
+		var ctxWithContainer context.Context
+
+		if ctxWithContainer, err = appctx.CopyContainer(ctx, context.Background()); err != nil {
```

Comment:
Not sure what I think about overwriting the previous error, doesn't matter I guess because this error here should never happen...